### DISCUSSION TITLE
Keycloak separate database for separate realms - Multitenancy

### DISCUSSION THREAD

**Discussion by souravs17031999** [keycloak/keycloak#17576] [Archive]
Hi community, 

As far as I understand Keycloak currently supports multitenancy at Database level but within the same database architechture pattern (by having identifiers columns here its' realmId), is there a way to configure storage layer in such a way that we can have separate databases per realm level so to say to go towards hard multitenancy approach ?

**Comment by souravs17031999** [comment]
Any ideas ? @mposolda @pedroigor @ssilvert @stianst

**Comment by hmlnarik** [comment]
This is expected to be delivered with the new store in #19369.

**Reply by souravs17031999** [reply]
Thanks @hmlnarik for the info !
